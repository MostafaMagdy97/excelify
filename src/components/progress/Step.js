// @flow
import React from 'react';
import styled from 'styled-components';
import { Flex } from 'grid-styled';
import { bgColor } from 'styled-system';

import Text from '../text/Text';
import { COLORS, COLORS_VALUES } from '../../base/Colors';
import { FONT_WEIGHTS, FONT_TYPES } from '../../base/Typography';

const pointRadius = '12px';

const Point = styled.div`
  ${bgColor};
  max-height: ${pointRadius};
  max-width: ${pointRadius};
  min-height: ${pointRadius};
  min-width: ${pointRadius};
  border-radius: 50%;
`;

const Line = styled.div`
  ${bgColor};
  border-radius: 15px;
  width: 102%;
  height: 4px;
  margin-right: -4px;
  margin-left: -4px;
`;

export type Props = {
  isDisabled: boolean,
  title: string,
  desc: string,
};

const Step = (props: Props) => {
  const { isDisabled, title, desc } = props;
  const color = isDisabled ? COLORS_VALUES[COLORS.DISABLED] : COLORS_VALUES[COLORS.BLUE];

  return (
    <Flex flexDirection="column" width={1}>
      <Flex flexDirection="row" alignItems="center">
        <Point bg={color} />
        <Line bg={color} />
      </Flex>

      <Flex flexDirection="column" mt={2} ml="6px">
        <Text type={FONT_TYPES.SUBHEADING} fontWeight={FONT_WEIGHTS.SEMI_BOLD} color={color}>
          {title}
        </Text>
        <Text type={FONT_TYPES.CAPTION} color={color}>
          {desc}
        </Text>
      </Flex>
    </Flex>
  );
};

export default Step;
