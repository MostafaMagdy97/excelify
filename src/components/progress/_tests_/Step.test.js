import React from 'react';
import { configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-enzyme';
import 'jest-styled-components';

import Step from '../Step';

configure({ adapter: new Adapter() });

describe('Step', () => {
  it('should render when not disabled', () => {
    const tree = render(<Step isDisabled={false} title="Title" desc="Desc" />);
    expect(tree).toMatchSnapshot();
  });

  it('should render when disabled', () => {
    const tree = render(<Step isDisabled title="Title" desc="Desc" />);
    expect(tree).toMatchSnapshot();
  });
});
