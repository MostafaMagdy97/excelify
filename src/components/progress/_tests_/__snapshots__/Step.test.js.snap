// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step should render when disabled 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  box-sizing: border-box;
  margin-left: 6px;
  margin-top: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  color: #637282;
  font-weight: 600;
  font-size: 17px;
  line-height: 24px;
  font-display: fallback;
}

.c6 {
  color: #637282;
  font-weight: 500;
  font-size: 13px;
  line-height: 24px;
  font-display: fallback;
}

.c2 {
  background-color: #637282;
  max-height: 12px;
  max-width: 12px;
  min-height: 12px;
  min-width: 12px;
  border-radius: 50%;
}

.c3 {
  background-color: #637282;
  border-radius: 15px;
  width: 102%;
  height: 4px;
  margin-right: -4px;
  margin-left: -4px;
}

@media screen and (min-width:40em) {
  .c5 {
    font-size: 17px;
  }
}

@media screen and (min-width:52em) {
  .c5 {
    font-size: 16px;
  }
}

@media screen and (min-width:64em) {
  .c5 {
    font-size: 16px;
  }
}

@media screen and (min-width:40em) {
  .c6 {
    font-size: 13px;
  }
}

@media screen and (min-width:52em) {
  .c6 {
    font-size: 12px;
  }
}

@media screen and (min-width:64em) {
  .c6 {
    font-size: 12px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      
    </div>
    <div
      class="c3"
    >
      
    </div>
  </div>
  <div
    class="c4"
  >
    <label
      class="c5"
      color="#637282"
      font-size="17px,17px,16px,16px"
      font-weight="600"
      lineheight="24px"
      tag="label"
      type="subheading"
    >
      Title
    </label>
    <label
      class="c6"
      color="#637282"
      font-size="13px,13px,12px,12px"
      font-weight="500"
      lineheight="24px"
      tag="label"
      type="caption"
    >
      Desc
    </label>
  </div>
</div>
`;

exports[`Step should render when not disabled 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  box-sizing: border-box;
  margin-left: 6px;
  margin-top: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  color: #0070E0;
  font-weight: 600;
  font-size: 17px;
  line-height: 24px;
  font-display: fallback;
}

.c6 {
  color: #0070E0;
  font-weight: 500;
  font-size: 13px;
  line-height: 24px;
  font-display: fallback;
}

.c2 {
  background-color: #0070E0;
  max-height: 12px;
  max-width: 12px;
  min-height: 12px;
  min-width: 12px;
  border-radius: 50%;
}

.c3 {
  background-color: #0070E0;
  border-radius: 15px;
  width: 102%;
  height: 4px;
  margin-right: -4px;
  margin-left: -4px;
}

@media screen and (min-width:40em) {
  .c5 {
    font-size: 17px;
  }
}

@media screen and (min-width:52em) {
  .c5 {
    font-size: 16px;
  }
}

@media screen and (min-width:64em) {
  .c5 {
    font-size: 16px;
  }
}

@media screen and (min-width:40em) {
  .c6 {
    font-size: 13px;
  }
}

@media screen and (min-width:52em) {
  .c6 {
    font-size: 12px;
  }
}

@media screen and (min-width:64em) {
  .c6 {
    font-size: 12px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      
    </div>
    <div
      class="c3"
    >
      
    </div>
  </div>
  <div
    class="c4"
  >
    <label
      class="c5"
      color="#0070E0"
      font-size="17px,17px,16px,16px"
      font-weight="600"
      lineheight="24px"
      tag="label"
      type="subheading"
    >
      Title
    </label>
    <label
      class="c6"
      color="#0070E0"
      font-size="13px,13px,12px,12px"
      font-weight="500"
      lineheight="24px"
      tag="label"
      type="caption"
    >
      Desc
    </label>
  </div>
</div>
`;
